src/main/java/io/cryostat/net/OpenShiftAuthManager.java

```diff
@@ -78,65 +92,97 @@ public AuthenticationScheme getScheme() {
     }
 
     @Override
-    public Future<Boolean> validateToken(Supplier<String> tokenProvider) {
+    public Future<Boolean> validateToken(
+            Supplier<String> tokenProvider, Set<ResourceAction> resourceActions) {
         String token = tokenProvider.get();
         if (StringUtils.isBlank(token)) {
             return CompletableFuture.completedFuture(false);
         }
-        return CompletableFuture.supplyAsync(
-                        () -> {
-                            AuthRequest evt = new AuthRequest();
-                            evt.begin();
+        if (resourceActions.isEmpty()) {
+            return CompletableFuture.completedFuture(true);
```

Comment:
Good catch, this initially just happened to map to requests like `GET /notifications_url`, but there are now and always was the possibility of other requests that don't necessarily have a ResourceAction but should still require authentication.

If there is no associated resource model, how should the RBAC permissions for the given token be checked? Should this instead check for a permission that should normally always be granted to users - something again like the previous "list routes"?

Reply:
```diff
@@ -99,7 +100,8 @@ public class OpenShiftAuthManager extends AbstractAuthManager {
             return CompletableFuture.completedFuture(false);
         }
         if (resourceActions.isEmpty()) {
-            return CompletableFuture.completedFuture(true);
+            resourceActions = EnumSet.copyOf(resourceActions);
+            resourceActions.add(ResourceAction.READ_TARGET);
         }
 
         try (OpenShiftClient client = clientProvider.apply(token)) {
```

Or even just this, it's a pretty simple workaround and asserts that the user provided a valid token and has at least a bare-minimum permission that would be expected. If the user doesn't have READ_TARGET then they can't really do anything else with Cryostat, either.

Reply:
True. There's also this API: https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-review-v1/

You pass it the token and it will return if it authenticated and who it belongs to.